etsin_finder/frontend/js/components/dataset/formatChanger.jsx

```diff
@@ -0,0 +1,128 @@
+{
+  /**
+   * This file is part of the Etsin service
+   *
+   * Copyright 2017-2018 Ministry of Education and Culture, Finland
+   *
+   *
+   * @author    CSC - IT Center for Science Ltd., Espoo Finland <servicedesk@csc.fi>
+   * @license   MIT
+   */
+  }
+
+  import React, { Component } from 'react'
+  import PropTypes from 'prop-types'
+  import { withRouter } from 'react-router-dom'
+  import { withTheme } from 'styled-components'
+  import ErrorPage from '../errorpage'
+  import DatasetQuery from '../../stores/view/datasetquery'
+  import Accessibility from '../../stores/view/accessibility'
+
+
+  import FormatSelect from './formatselect'
+
+
+  class FormatChanger extends Component {
+    constructor(props) {
+      super(props)
+
+      this.state = {
+        formats: [],
+        error: false,
+        selected: '',
+      }
+
+      this.query = this.query.bind(this)
+    }
+
+    componentDidMount() {
+      Accessibility.resetFocus()
+      this.query()
+    }
+
+    checkFields = (data) => {
+      const rd = data.research_dataset;
+      let dataciteExists = false;
+      let fields = {};
+
+      // Check that doi exists in one of the identifiers
+      if ((typeof data.preservation_identifier !== 'undefined' && data.preservation_identifier.includes('doi'))
+      || (typeof rd.preferred_identifier !== 'undefined' && rd.preferred_identifier.includes('doi'))) {
+        dataciteExists = true;
+      }
+
+      if (dataciteExists) {
+        fields = [{ label: 'Metax JSON', value: 'metax' },
+        { label: 'Datacite without validation', value: 'fairdata_datacite' },
+        { label: 'Datacite format', value: 'datacite' }]
+      } else {
+        fields = [{ label: 'Metax JSON', value: 'metax' }]
+      }
+      this.setState({
+        formats: fields,
+        error: false,
+        selected: '',
+      })
+    }
+
+
+    changeFormat = (name, format) => {
+      this.setState(
+        {
+          selected: format,
+          url: `https://metax-test.csc.fi/rest/datasets/${this.props.idn}?dataset_format=${format.value}`
+        },
+        () => {
+          this.openFormat(this.state.url)
+        }
+      )
+    }
+
+    openFormat = (url) => {
+      const win = window.open(url, '_blank');
+      win.focus();
+    }
+
+    query(customId) {
```

Comment:
The query shouldn't be done in this component when it is done in the parent and stored in the mobX state. Rather than querying it here every time the component or its parent refreshes, get the value from the `@observable results = []` in `/stores/view/datasetquery.DatasetQuery`.